lambdaNotes/src/components/Note.js

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import axios from "axios";
+import EditNote from "./EditNote";
+import "../App.css";
+import {
+  LeftBar,
+  StyledLink,
+  ModalH3,
+  TitleH1,
+  NoteCardContainer,
+  NoteTitle,
+  NoteP,
+  EditNoteLink,
+  DeleteNoteLink,
+  SignOut
+} from "../Styles";
+
+const baseUrl = process.env.NODE_ENV === "development" ? "http://localhost:5000" : "";
+
+class Note extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editing: false,
+      title: this.props.notes.title,
+      content: this.props.notes.content
+    };
+  }
+
+  editNote = event => {
+    event.preventDefault();
+    this.setState({ editing: true });
+  };
+
+  deleteNote = event => {
+    event.preventDefault();
+    const url = `${baseUrl}/notes/${
+      this.state.note.note[0].id
+    }`;
+
+    axios
+      .delete(url)
+      .then(response => {
+        console.log(response);
+        this.props.updateDeleted(this.state.note.note[0].id);
+      })
+      .catch(error => {
+        console.error(error);
+      });
+    this.props.history.push("/");
+  };
+
+  deleteModal = () => {
+    console.log("button clicked");
+    document.querySelector(".modal").classList.toggle("show");
+  };
+
+  componentDidMount(props) {
+    const id = this.props.match.params.id;
+    this.fetchNote(id);
+  }
+
+  fetchNote = id => {
+    // console.log(this.state.id);
+    axios
+      .get(`${baseUrl}/notes/${id}`)
+      .then(response => {
+        this.setState(() => ({ note: response.data }));
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  };
+
+  componentWillReceiveProps(newProps) {
```

Comment:
generally it's preferred to use componentDidUpdate rather than componentWillReceiveProps